components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -64,6 +64,79 @@ auto cmvdc (const ViewT& v_d) -> typename ViewT::HostMirror {
   return v_h;
 }
 
+template<typename DeviceT>
+void run_scalar_valued_fns()
+{
+  /*
+  Most common physics functions operate simultaneously on packs of vertical indices. The functions tested here
```

Comment:
To be clear, the common physics functions come in two flavors: one templated on ScalarT, without assumptions on what that scalar is; and one that works on a full column of ScalarT (which simply runs a ||for loop calling the former). As such, one can call them with `Scalar=Pack<Real,N>` (as we often do), as well as with `Scalar=Real`.

That said, since these functions are always gonna be called with 2d inputs/outputs, it's perfectly fine to test them separately.

Reply:
The new functions added here are different than other common physics functions because they *only* operate on reals. It would be nonsensical to test them on packs because all common physics functions are called within a loop over the horizontal and these functions are also only defined for a single vertical layer. I originally had these functions templated on ScalarT but then I got worried that someone might test them on Packs, get a fail (because what does a pack even mean in this context), and spend a lot of time debugging.